### DISCUSSION TITLE
Project not reaching proper framerate on Nintendo Switch with only vegetation

### DISCUSSION THREAD

**Discussion by Rodrigo-Cuadriello** [visualdesigncafe/nature-renderer#60] [Help]
Hi, please if you could help me out figuring why a terrain with only grass has a dropframe of nearly 8 fps on the Nintendo Switch Devkit, I've integrated the Nature Render 2022 in Unity 2022.3.33, added the Nature Renderer component to my terrain, fixed the shaders for the grass, and placed some grass. When I'm testing on my Nintendo Switch devkit the game runs ok, with sections up to 30fps, but when I get near the grass the framerate drops down to 8 fps. I'm using URP, here are some screenshots form Unity and the Nature Render Profiler.

![Captura3](https://github.com/visualdesigncafe/nature-renderer/assets/64656681/a92f19ba-0c2b-4c91-abc7-87b15dd46eb0)
![2](https://github.com/visualdesigncafe/nature-renderer/assets/64656681/fb20301b-c74f-4ef0-8e3a-43dc0a87bd00)
![3](https://github.com/visualdesigncafe/nature-renderer/assets/64656681/e0ebef20-9152-4864-b4d0-af6b6157ecff)
![4](https://github.com/visualdesigncafe/nature-renderer/assets/64656681/b477bef6-a1ba-41e3-a04e-45b6a3e53684)

**Comment by maxplooi** [comment]
1. It looks like the default Unity renderer is still enabled, so all the grass is rendered twice. Please make sure that "**Draw**" is disabled in the **Tree & Detail Objects** section of the terrain settings.
2. If the performance only gets bad when being near to the grass then the performance issue is usually caused by the shaders/overdraw. Most likely, the shaders are too complex or the amount of [overdraw](https://visualdesigncafe.com/nature-renderer/docs/2020/overdraw) is too high (or both). You can try reducing the [density](https://visualdesigncafe.com/nature-renderer/docs/2022/density) of the grass to improve performance (either reduce the density everywhere, or use the **Density in Distance** setting to dynamically decrease the density). The easiest "fix" for the performance is to select a simpler shader that uses less complex lighting.
